---
title: "A basic workflow for tissue specific deconvolution model"
author: "Ziyi Chen"
output: html_document

abstract:
Estimate the relative proportion of tissue immune cell from tissue transcriptome with a tissue specific model.
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Introduction
The tissue specific computational model can be used to predict the relative proportion of immune cell with a series of tissue specific training signature matrix.

It is achieved by three major steps:
1. Raw bulk RNA-Seq data preprocessing
2. Immune cell proportion prediction

# Raw bulk RNA-Seq data preprocessing
Raw fastq format sequencing data should be first preprocessed into the expression matrix and six major steps were included. 
(1)	Quality control;
(2)	Mapping. Mapping reads into the reference genome with STAR;
(3)	Sorting. Sort the mapped reads according to their name;
(4)	Strand. Get the strand information for the library;
(5)	Quantification. Quantify Gene expression with HTSeq.
(6)	Normalization. Subtypes of T cell receptor and B cell receptors were merged into major families and the raw expression matrix were normalized with the quantile normalization method.
The shell script to preprocess it can be obtained from https://github.com/wuaipinglab/ImmuCC/blob/master/webserver/RNASeq_pipeline.sh.

sh RNASeq_pipeline.sh `${Directory to the base}` PE `${Directory to the software}` `${Directory to the reference}` `${Directory to the scripts}` 24
`${Directory to the base}`: This directory contains 7 files, namely, 01fastq, 02trimmed, 03mapping, 04sorted, 05htseq, raw_fastqc, new_fastqc. 
`${Directory to the software}`: Softwares used were all in this directory including: FastQC, STAR, samtools, RSeQC, htseq-count, R.
`${Directory to the reference}`: This directory contains all reference data used. In my analysis, the following reference data including: Mus_musculus.GRCm38.83.gtf, Mus_musculus.GRCm38.dna.primary_assembly.83.fa, GRCm38_mm10_Ensembl.bed were used. You can download the lattest version as you want.
`${Directory to the scripts}`: Both shell scripts and R scripts used including: 02qc.sh, 03mapping.sh, 04samtools.sh, 05-1.strand.sh, 05-2.RSEQc.stat.R, 06htseq.sh, MouseHTSeq_counts_stat.Rwere all put in this directory.

## Immune cell proportion prediction
The CRAN package `“e1071”` were used to conduct SVR analysis.The Bioconductor package `“preprocessCore”` were used to

E.g. To estimate the immune cell proportion from the transcriptome of lung, the lung specific signature matrix was used to normaliaze the transcriptome data. As the transcriptome data have been normaliazed when we preprocess the raw sequencing data, it will be not necessary to normalize it again.

use the function `ImmuCC` to get the relative proportion of immune cells
Immune.proportion <- ImmuCC (expression, training_data = ”Lung.sig.matrix.csv”)

You can visualiaze the result with function `pheatmap` in package `pheatmap`.
pheatmap(Immune.proportion)

